tests/unit/test_charm.py

```diff
@@ -219,10 +251,100 @@ def test_report_visits_by_ip_empty(
         assert: there is nothing to show
         """
         mock_pull.return_value = io.StringIO("")
-        action = self.harness.charm._report_visits_by_ip(mock.MagicMock())
+        action = self.harness.charm._report_visits_by_ip()
         expected = []
         self.assertEqual(action, expected)
 
+    def test_get_ip_multiple_ips(
```

Comment:
Nice additions, can you think of a way to simplify the tests using https://docs.pytest.org/en/6.2.x/parametrize.html?

Reply:
We could use parametrization, but using unittest.testCase in our test class would give us a problem with that... e.g. https://stackoverflow.com/questions/35561843/py-test-parametrize-test-cases-from-classes Should I get rid of the testCase, change the assertions and add parameters anyway?

Reply:
As discussed, we'll park parametrization and put a story on our backlog to convert these to pytest